### PR TITLE
[Doc] Improve S3File Source & S3File Sink document

### PR DESCRIPTION
**Pull request by EricJoy2048**
sub task of https://github.com/apache/seatunnel/issues/4572

### CODE REVIEW

File: docs/en/connector-v2/sink/S3File.md [hunk 1]
```diff
@@ -30,60 +23,106 @@ By default, we use 2PC commit to ensure `exactly-once`
...
+| field_delimiter                  | string  | no       | '\001'                                                | Only used when file_format is text                                                                                                                                    |
+| row_delimiter                    | string  | no       | "\n"                                                  | Only used when file_format is text                                                                                                                                    |
+| have_partition                   | boolean | no       | false                                                 | Whether you need processing partitions.                                                                                                                               |
+| partition_by                     | array   | no       | -                                                     | Only used then have_partition is true                                                                                                                                 |
```
Comment (TaoZex) [inline docs/en/connector-v2/sink/S3File.md]:
```suggestion
| partition_by                     | array   | no       | -                                                     | Only used when have_partition is true                                                                                                                                 |
```

File: docs/en/connector-v2/sink/S3File.md [hunk 2]
```diff
@@ -30,60 +23,106 @@ By default, we use 2PC commit to ensure `exactly-once`
...
+| partition_dir_expression         | string  | no       | "${k0}=${v0}/${k1}=${v1}/.../${kn}=${vn}/"            | Only used then have_partition is true                                                                                                                                 |
```
Comment (TaoZex) [inline docs/en/connector-v2/sink/S3File.md]:
```suggestion
| partition_dir_expression         | string  | no       | "${k0}=${v0}/${k1}=${v1}/.../${kn}=${vn}/"            | Only used when have_partition is true                                                                                                                                 |
```

File: docs/en/connector-v2/sink/S3File.md [hunk 3]
```diff
@@ -30,60 +23,106 @@ By default, we use 2PC commit to ensure `exactly-once`
...
+| is_partition_field_write_in_file | boolean | no       | false                                                 | Only used then have_partition is true                                                                                                                                 |
```
Comment (TaoZex) [inline docs/en/connector-v2/sink/S3File.md]:
```suggestion
| is_partition_field_write_in_file | boolean | no       | false                                                 | Only used when have_partition is true                                                                                                                                 |
```

**Review by TaoZex — APPROVED** [review]
LGTM
It would be better to fix these typos.

**Comment by EricJoy2048** [comment]
> LGTM It would be better to fix these typos.

Thanks, I finished it.

**Review by TaoZex — APPROVED** [review]
+1

**Comment by ic4y** [comment]
@EricJoy2048 Conflicts need to be resolved.

**Comment by EricJoy2048** [comment]
> @EricJoy2048 Conflicts need to be resolved.

done

File: docs/en/connector-v2/sink/S3File.md [hunk 4]
```diff
@@ -208,6 +242,70 @@ Writer the sheet of the workbook
...
+}
+
+source {
+  # This is a example source plugin **only for test and demonstrate the feature source plugin**
+  FakeSource {
+    parallelism = 1
+    result_table_name = "fake"
+    row.num = 16
+    schema = {
+      fields {
```
Comment (zhilinli123) [inline docs/en/connector-v2/sink/S3File.md:265]: Add all types of tests
reference: https://github.com/apache/seatunnel/pull/5132/files

File: docs/en/connector-v2/source/S3File.md [hunk 5]
```diff
@@ -279,9 +239,30 @@ Reader the sheet of the workbook,Only used when file_format_type is excel.
...
   }
+}
 
+transform {
+  # If you would like to get more information about how to configure seatunnel and see full list of transform plugins,
+    # please go to https://seatunnel.apache.org/docs/category/transform-v2
+}
+
+sink {
+    jdbc {
```
Comment (hailin0) [inline docs/en/connector-v2/source/S3File.md]: use console sink？

File: docs/en/connector-v2/source/S3File.md [hunk 6]
```diff
@@ -300,9 +281,56 @@ Reader the sheet of the workbook,Only used when file_format_type is excel.
...
+  }
+}
+
+transform {
+  # If you would like to get more information about how to configure seatunnel and see full list of transform plugins,
+    # please go to https://seatunnel.apache.org/docs/category/transform-v2
+}
+
+sink {
+    jdbc {
```
Comment (hailin0) [inline docs/en/connector-v2/source/S3File.md]: use console sink？

Comment (EricJoy2048) [reply docs/en/connector-v2/source/S3File.md]: @hailin0  JDBC Sink is a more common usage method compared to Console Sink, isn't it?

File: docs/en/connector-v2/sink/S3File.md (hunk 4 above)
Comment (EricJoy2048) [reply docs/en/connector-v2/sink/S3File.md:265]: done

File: docs/en/connector-v2/source/S3File.md (hunk 6 above)
Comment (hailin0) [reply docs/en/connector-v2/source/S3File.md]: What if the user only uses s3 file without installing the database

Comment (hailin0) [reply docs/en/connector-v2/source/S3File.md]: Or when the jdbc connector is updated, who can remember that it needs to be updated

**Review by ic4y — APPROVED** [review]

**Review by liugddx — APPROVED** [review]

**Pull request merged**